examples/bzlmod/pip_repository_annotations_test.py

```diff
@@ -0,0 +1,129 @@
+        return env
+
+    def test_build_content_and_data(self):
+        r = runfiles.Create()
+        rpath = r.Rlocation(
+                "rules_python~override~pip~{}/generated_file.txt".format(
+                    self.wheel_pkg_dir(),
+                    ),
+                )
+        generated_file = Path(rpath)
+        self.assertTrue(generated_file.exists())
+
+        content = generated_file.read_text().rstrip()
+        self.assertEqual(content, "Hello world from build content file")
+
+    def test_copy_files(self):
+        r = runfiles.Create()
+        rpath = r.Rlocation(
+                "rules_python~override~pip~{}/copied_content/file.txt".format(
+                    self.wheel_pkg_dir()
+                    )
+                )
+        copied_file = Path(rpath)
+        self.assertTrue(copied_file.exists())
+
+        content = copied_file.read_text().rstrip()
+        self.assertEqual(content, "Hello world from copied file")
+
+    def test_copy_executables(self):
+        r = runfiles.Create()
+        rpath = r.Rlocation(
+                "rules_python~override~pip~{}/copied_content/executable{}".format(
+                    self.wheel_pkg_dir(),
+                    ".exe" if platform.system() == "windows" else ".py",
+                    )
+                )
+        executable = Path(rpath)
+        self.assertTrue(executable.exists())
+
+        proc = subprocess.run(
+            [sys.executable, str(executable)],
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        stdout = proc.stdout.decode("utf-8").strip()
+        self.assertEqual(stdout, "Hello world from copied executable")
+
+    def test_data_exclude_glob(self):
+        current_wheel_version = "0.40.0"
+
+        r = runfiles.Create()
+        dist_info_dir = "rules_python~override~pip~{}/site-packages/wheel-{}.dist-info".format(
+                self.wheel_pkg_dir(),
+                current_wheel_version,
+                )
+
+        # Note: `METADATA` is important as it's consumed by https://docs.python.org/3/library/importlib.metadata.html
+        # `METADATA` is expected to be there to show dist-info files are included in the runfiles.
+        metadata_path = r.Rlocation("{}/METADATA".format(dist_info_dir))
+
+        # However, `WHEEL` was explicitly excluded, so it should be missing
+        wheel_path = r.Rlocation("{}/WHEEL".format(dist_info_dir))
+
+        # Because windows does not have `--enable_runfiles` on by default, the
+        # `runfiles.Rlocation` results will be different on this platform vs
+        # unix platforms. See `@rules_python//python/runfiles` for more details.
+        if platform.system() == "Windows":
+            self.assertTrue(Path(metadata_path).exists())
+            self.assertFalse(Path(wheel_path).exists())
+        else:
+            self.assertTrue(Path(metadata_path).exists())
+            self.assertFalse(Path(wheel_path).exists())
```

Comment:
These code paths look the same? Am i not seeing the diff?

Reply:
Yep. I made some tweaks. Will fix